tests/page-objects/dataTable.ts

```diff
@@ -0,0 +1,77 @@
+import { Page, Locator, expect } from "@playwright/test"
+import { Dialog } from "puppeteer";
+
+export class dataTableElements{
+    protected page: Page;
+    readonly dataTable
+    readonly table: Map<number,{editBtn:Locator,deleteBtn:Locator,nameEdit:Locator,emailEdit:Locator,saveBtn:Locator, name:Locator, email:Locator}>
+   
+    constructor(page: Page) {
+        this.page = page;
+        this.dataTable = this.page.getByTestId('dataTable');
+    
+        this.table = new Map([
+          [1, {
+            editBtn: this.page.getByTestId('editBtn1'),
+            deleteBtn: this.page.getByTestId('deleteBtn1'),
+            nameEdit : this.page.locator('//input[@class="edit-name"]'),
+            emailEdit : this.page.locator('//input[@class="edit-email"]'),
+            saveBtn : this.page.getByText('Save'),
+            name : this.page.getByTestId('name1'),
+            email : this.page.getByTestId('email1')
+          }],
+          [2, {
+            editBtn: this.page.getByTestId('editBtn2'),
+            deleteBtn: this.page.getByTestId('deleteBtn2'),
+            nameEdit : this.page.locator('//input[@class="edit-name"]'),
+            emailEdit : this.page.locator('//input[@class="edit-email"]'),
+            saveBtn : this.page.getByText('Save'),
+            name : this.page.getByTestId('name2'),
+            email : this.page.getByTestId('email2')
+          }],
+          [3, {
+            editBtn: this.page.getByTestId('editBtn3'),
+            deleteBtn: this.page.getByTestId('deleteBtn3'),
+            nameEdit : this.page.locator('//input[@class="edit-name"]'),
+            emailEdit : this.page.locator('//input[@class="edit-email"]'),
+            saveBtn : this.page.getByText('Save'),
+            name : this.page.getByTestId('name3'),
+            email : this.page.getByTestId('email3')
+          }]
+        ]);
+    }
+    async verifyTableSectionsIsDisplayed(){
+        await expect(this.dataTable).toBeVisible();
+    }
+    async clickEditButtonandVerifyIsEditable(rowNumber:number, newName:string, newEmail:string ): Promise <void>{
```

Comment:
Use the arguments of a function to inform the naming of the function. You are using name, email and index to click an edit button and verify 'is editable'. What table is it referring to? What is 'editable'?